### ISSUE TITLE
How do I calculate FPS on my validation set

### ISSUE THREAD

**Issue by Wanghe1997** [ultralytics/yolov5#6873]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I divided my dataset into training set and verification set (test set) in a ratio of 7:3, initialized training on my own training set, and got the trained weight. I want to derive the FPS of my model on my validation set. how do I do that? Thanks

**Comment by glenn-jocher** [comment]
@Wanghe1997 Frames Per Second (FPS) is just 1 second divided by the inference time in seconds.

**Comment by Wanghe1997** [comment]
> @Wanghe1997 Frames Per Second (FPS) is just 1 second divided by the inference time in seconds.
![结果](https://user-images.githubusercontent.com/45507837/156981446-53a523e7-5480-455b-80ae-5bb0a4ceef8c.png)
Here are the results of running detect.py to infer my own test set(570 images). The size of each image and the number of targets on my test set are not fixed, so the inference time is longer on some images and shorter on some images. Your program does not implement the function of calculating the average time of inference on the entire data set (it is recommended to add it later), because most people's test set is not only one picture, but a series of pictures. From this picture of mine, my problem is that I don't know which way to calculate fps is correct. The first: **570/30.931=18.428**. The second (take any picture): **1/0.009=111.11**. The results of these two calculation methods are quite different. Because you didn't write the result of the average inference time in your code, I can only see the inference time of each image (however, the size of each image is not necessarily the same, and the time is not the same, so I think this is not correct). So, in terms of my results for this image, is the fps **18.428** or **111.11**? thanks

**Comment by glenn-jocher** [comment]
@Wanghe1997 average speeds are clearly displayed. 

If not your code is very out of date.

**Comment by Wanghe1997** [comment]
> @Wanghe1997 average speeds are clearly displayed.
> 
> If not your code is very out of date.

i use v5.0 code. And where is the average speeds in my picture? I only see the inference time for a single image, there is no average time in the graph. And why is 30.931/570=0.05426491228 seconds, which is so different from the 0.009 seconds of the single picture shown in the figure?
